MIG-1453: updating upgrade documentation (#65453)
openshift/openshift-docs: modules/migration-upgrading-mtc-on-ocp-4.adoc

You can upgrade the {mtc-full} ({mtc-short}) on {product-title} {product-version} by using the Operator Lifecycle Manager.

[IMPORTANT]
.Migration paths
====
* Migrating from {product-title} 3 to {product-title} 4 requires a legacy {mtc-short} Operator and {mtc-short} 1.7.x.
* Migrating from {mtc-short} 1.7.x to {mtc-short} 1.8.x is not supported.
* You must use {mtc-short} 1.7.x to migrate anything with a source of {product-title} 4.9 or earlier.

> ** {mtc-short} 1.7.z must be used on both source and destination.

Reviewer: We missed a dot-z to dot-x in our last iteration. :shushing_face: 

```suggestion
** {mtc-short} 1.7.x must be used on both source and destination.
```

> ** {mtc-short} 1.7.z must be used on both source and destination.
> * MTC 1.8.x only supports migrations from {product-title} 4.10 or later to {product-title} 4.10 or later. For migrations only involving cluster versions 4.10 and later, either 1.7.x or 1.8.x may be used. However, but it must be the same MTC 1.Y.z on both source & destination.
> ** Migration from source {mtc-short} 1.7.z to destination {mtc-short} 1.8.z is unsupported.
> ** Migration from source {mtc-short} 1.8.z to destination {mtc-short} 1.7.z is unsupported.
> ** Migration from source {mtc-short} 1.7.z to destination {mtc-short} 1.7.z is supported.
> ** Migration from source {mtc-short} 1.8.z to destination {mtc-short} 1.8.z is supported

Reviewer: Docs already use dot-x throughout.

```suggestion
** {mtc-short} 1.7.z must be used on both source and destination.
* MTC 1.8.x only supports migrations from {product-title} 4.10 or later to {product-title} 4.10 or later. For migrations only involving cluster versions 4.10 and later, either 1.7.x or 1.8.x may be used. However, it must be the same MTC version on both source & destination.
** Migration from source {mtc-short} 1.7.x to destination {mtc-short} 1.8.x is unsupported.
** Migration from source {mtc-short} 1.8.x to destination {mtc-short} 1.7.x is unsupported.
** Migration from source {mtc-short} 1.7.x to destination {mtc-short} 1.7.x is supported.
** Migration from source {mtc-short} 1.8.x to destination {mtc-short} 1.8.x is supported
```

For "_must be the same MTC 1.Y.z on both source & destination_", I think it'll be cleaner and with this suggestion, but rework that as necessary if it isn't sufficient.  :sunglasses:

> [IMPORTANT]
> .Migration paths
> ====
> * Migrating from {product-title} 3 to {product-title} 4 requires a legacy {mtc-short} Operator and {mtc-short} 1.7.x.
> * Migrating from {mtc-short} 1.7.x to {mtc-short} 1.8.x is not supported.
> * You must use {mtc-short} 1.7.x to migrate anything with a source of {product-title} 4.9 or earlier.
> ** {mtc-short} 1.7.x must be used on both source and destination.
> * MTC 1.8.x only supports migrations from {product-title} 4.10 or later to {product-title} 4.10 or later. For migrations only involving cluster versions 4.10 and later, either 1.7.x or 1.8.x may be used. However, it must be the same MTC version on both source & destination.
> ** Migration from source {mtc-short} 1.7.x to destination {mtc-short} 1.8.x is unsupported.
> ** Migration from source {mtc-short} 1.8.x to destination {mtc-short} 1.7.x is unsupported.
> ** Migration from source {mtc-short} 1.7.x to destination {mtc-short} 1.7.x is supported.
> ** Migration from source {mtc-short} 1.8.x to destination {mtc-short} 1.8.x is supported
> ====

Reviewer 2: Understand that this is critical information, but in general we try to avoid extensive styling within admonition blocks. Could the admonition be reduced to a warning that the user must review the supported migration paths, and the bulleted list moved out immediately after like so?

```suggestion
[IMPORTANT]
====
When upgrading the{mtc-short} by using the Operator Lifecycle Manager, you must use a supported migration path.
====

.Migration paths
* Migrating from {product-title} 3 to {product-title} 4 requires a legacy {mtc-short} Operator and {mtc-short} 1.7.x.
* Migrating from {mtc-short} 1.7.x to {mtc-short} 1.8.x is not supported.
* You must use {mtc-short} 1.7.x to migrate anything with a source of {product-title} 4.9 or earlier.
** {mtc-short} 1.7.x must be used on both source and destination.
* MTC 1.8.x only supports migrations from {product-title} 4.10 or later to {product-title} 4.10 or later. For migrations only involving cluster versions 4.10 and later, either 1.7.x or 1.8.x may be used. However, it must be the same MTC version on both source & destination.
** Migration from source {mtc-short} 1.7.x to destination {mtc-short} 1.8.x is unsupported.
** Migration from source {mtc-short} 1.8.x to destination {mtc-short} 1.7.x is unsupported.
** Migration from source {mtc-short} 1.7.x to destination {mtc-short} 1.7.x is supported.
** Migration from source {mtc-short} 1.8.x to destination {mtc-short} 1.8.x is supported
```

You can upgrade the {mtc-full} ({mtc-short}) on {product-title} {product-version} by using the Operator Lifecycle Manager.

[IMPORTANT]
====

> When upgrading the{mtc-short} by using the Operator Lifecycle Manager, you must use a supported migration path.

Reviewer: ```suggestion
When upgrading the {mtc-short} by using the Operator Lifecycle Manager, you must use a supported migration path.
```
